core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -593,4 +596,278 @@ class SocketServerTest extends JUnitSuite {
+      testableServer.shutdown()
+      testableServer.metrics.close()
+    }
+  }
+
+  @Test
+  def processDisconnectedException(): Unit = {
+    selectorException(SelectorOperation.DISCONNECTED)
+  }
+
+  @Test
+  def pollException(): Unit = {
+    selectorException(SelectorOperation.POLL)
+  }
+
+  private def selectorException(operation: SelectorOperation.Value): Unit = {
+    val testableServer = startTestableServer()
+    try {
+      val socket = connect(testableServer)
+      sendRequest(socket, producerRequestBytes)
+      receiveRequest(testableServer.requestChannel)
+      val testableSelector = testableServer.testableSelector
+
+      testableSelector.failures += operation -> new IllegalStateException("Test exception")
+      socket.close()
+      testableSelector.operationCounts.clear()
+      testableSelector.waitForOperations(operation, 1)
+
+      assertProcessorHealthy(testableServer)
+    } finally {
+      testableServer.shutdown()
+      testableServer.metrics.close()
+    }
+  }
+
+  private def startTestableServer(): TestableSocketServer = {
+    props.put("listeners", "PLAINTEXT://localhost:0")
+    val testableServer = new TestableSocketServer
+    testableServer.startup()
+    testableServer
+  }
+
+  private def assertProcessorHealthy(testableServer: TestableSocketServer): Unit = {
+    val selector = testableServer.testableSelector
+    selector.failures.clear()
+    val socket = connect(testableServer)
+    val serializedBytes = producerRequestBytes
+    sendRequest(socket, serializedBytes)
+    val requestChannel = testableServer.requestChannel
+
+    val request = receiveRequest(requestChannel)
+    val connectionId = request.context.connectionId
+    processRequest(requestChannel, request)
+    assertEquals(serializedBytes.toSeq, receiveResponse(socket).toSeq)
+    assertNotNull("Channel should not have been closed", selector.channel(connectionId))
+    assertNull("Channel should not be closing", selector.closingChannel(connectionId))
+  }
+
+  class TestableSocketServer extends SocketServer(KafkaConfig.fromProps(props), new Metrics, Time.SYSTEM, credentialProvider) {
+
+    @volatile var selector: Option[TestableSelector] = None
+
+    override def newProcessor(id: Int, connectionQuotas: ConnectionQuotas, listenerName: ListenerName,
+                                protocol: SecurityProtocol, memoryPool: MemoryPool): Processor = {
+      new Processor(id, time, config.socketRequestMaxBytes, requestChannel, connectionQuotas,
+        config.connectionsMaxIdleMs, listenerName, protocol, config, metrics, credentialProvider, memoryPool) {
+        override protected[network] def createSelector(maxReceiveSize: Int, connectionMaxIdleMs: Long,
+             metrics: Metrics, time: Time, metricGrpPrefix: String, metricTags: java.util.Map[String, String],
+             metricsPerConnection: Boolean, recordTimePerConnection: Boolean,
+             channelBuilder: ChannelBuilder, memoryPool: MemoryPool): Selector = {
```

Comment:
Do we need all these parameters in `createSelector`?

Reply:
Removed parameters not required for test.